### PR TITLE
move resources into the individual controller package where they belong

### PR DESCRIPTION
**Pull request by mrIncompetent**
**What this PR does / why we need it**:
Move resources into the individual controller package where they belong.
Also this simplifies the DexCA function, which was more complex than actually needed.

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: api/pkg/controller/cluster/cluster_controller.go [hunk 1]
```diff
@@ -451,8 +451,10 @@ func (cc *Controller) processNextItem() bool {
 	}
 	defer cc.queue.Done(key)
 
-	if err := cc.syncCluster(key.(string)); err != nil {
-		cluster, errGetCluster := cc.clusterLister.Get(key.(string))
+	sKey := key.(string)
+	if err := cc.syncCluster(sKey); err != nil {
+		glog.Errorf("Failed syncinc '%s': %v", sKey, err)
```
Comment (alvaroaleman) [inline api/pkg/controller/cluster/cluster_controller.go:456]: Why do you re-introduce that?

Comment (alvaroaleman) [reply api/pkg/controller/cluster/cluster_controller.go:456]: And also, why not on  a high `V` level?

File: api/pkg/controller/cluster/resources.go [hunk 2]
```diff
@@ -231,7 +232,7 @@ func (cc *Controller) ensureSecrets(c *kubermaticv1.Cluster, data *resources.Tem
 
 	for _, op := range operations {
 		if err := resources.EnsureSecret(op.name, data, op.create, cc.secretLister.Secrets(c.Status.NamespaceName), cc.kubeClient.CoreV1().Secrets(c.Status.NamespaceName)); err != nil {
-			return fmt.Errorf("failed to ensure that the Secret exists: %v", err)
+			return fmt.Errorf("failed to ensure that the Secret '%s' exists: %v", op.name, err)
```
Comment (alvaroaleman) [inline api/pkg/controller/cluster/resources.go]: Can you reword this to `failed to ensure Secret '%s'`? The secret may exist, what failed is to ensure its desired state, not its existence

File: api/pkg/controller/cluster/cluster_controller.go (hunk 1 above)
Comment (mrIncompetent) [reply api/pkg/controller/cluster/cluster_controller.go:456]: I don't see any reason not to log an error during sync.
Otherwise i would need to watch the events.

We run into an error, so we log with `glog.Errorf`

File: api/pkg/controller/cluster/resources.go (hunk 2 above)
Comment (mrIncompetent) [reply api/pkg/controller/cluster/resources.go]: Done with https://github.com/kubermatic/kubermatic/pull/2506/commits/e81424bdd6a505a21bad3a88845a5e4758b0eb04

File: api/pkg/controller/cluster/cluster_controller.go (hunk 1 above)
Comment (alvaroaleman) [reply api/pkg/controller/cluster/cluster_controller.go:456]: But the error may be transient so if you handle a higher amount of clusters this may result in a lot of errors and these are debug logs that should not be turned on by default. Kubernetes controller-manager also does not log errors for any error in a sync loop it encounters

**Comment by mrIncompetent** [comment]
I'm closing this one and issue separate PR's
